runtime/misc.c

```diff
@@ -41,33 +44,53 @@ caml_timing_hook caml_minor_gc_end_hook = NULL;
 caml_timing_hook caml_finalise_begin_hook = NULL;
 caml_timing_hook caml_finalise_end_hook = NULL;
 
-#ifdef DEBUG
+#if defined(DEBUG) || defined(NATIVE_CODE)
```

Comment:
Is the `|| defined(NATIVE_CODE)` really necessary?  As far as I can see, `caml_failed_assert` is only called from the `CAMLassert` macro, and only if `DEBUG` is defined.  This is true in trunk as well as in the multicore-pr branch.